tests/testthat/test_posterior_predict.R

```diff
@@ -253,13 +253,21 @@ test_that("compatible with stan_lmer with offset", {
 })
 
 context("posterior_predict (stan_betareg)")
-test_that("compatible betareg", {
+test_that("compatible with stan_betareg with z", {
   data("GasolineYield", package = "betareg")
-  fit <- SW(stan_betareg(yield ~ temp, data = GasolineYield, 
-                     iter = ITER*5, chains = CHAINS, seed = SEED, refresh = REFRESH))
+  fit <- SW(stan_betareg(yield ~ batch + temp | temp, data = GasolineYield,
+                         iter = ITER, chains = CHAINS, seed = SEED, 
+                         refresh = REFRESH))
   check_for_error(fit)
   expect_linpred_equal(fit)
```

Comment:
I only got `posterior_predict` to work and forgot about `posterior_linpred`. I'm looking at what betareg does and it seems that `predict(betaregfit, type = "link")` (which I'm believe is what `posterior_linpred` is replicating) only returns the `mu` linear predictor even when `z` variables are declared. This doesn't really seem useful since the user can't really do anything with just `mu` if a `z` design matrix is also included in the model. Do we want both `posterior_linpred` and `linear.predictors` to return mu*phi?

```
fit <- SW(stan_betareg(yield ~ batch + temp | temp, data = GasolineYield,
                                               iter = ITER, chains = CHAINS, seed = SEED, 
                                               refresh = REFRESH))
check <- betareg::betareg(yield ~ batch + temp | temp, data = GasolineYield)

check_for_error(fit)
expect_linpred_equal(fit)

mu <- model.matrix(yield ~ batch + temp, data = GasolineYield) %*% coef(check)[1:11]
phi <- model.matrix(yield ~ temp, data = GasolineYield) %*% coef(check)[12:13]

mat <- matrix(
  cbind(
    apply(posterior_linpred(fit), 2, median),
    fit$linear.predictors,
    abs(apply(posterior_linpred(fit), 2, median)) - abs(fit$linear.predictors),
    predict(check, type = "link"),
    mu,
    phi,
    mu*phi
    ), nrow = 32
  )
colnames(mat) <- c("linpred", "linear.predictors", "diff", "betareg", "mu", "phi", "mu*phi")
mat

       linpred          linear.predictors   diff   betareg     mu          phi     mu*phi
 [1,]  0.822158294       0.738977686  0.083180607 -2.1975546 -2.1975546 4.351004  -9.5615692
 [2,]  0.588765080       0.529929302  0.058835778 -1.4723909 -1.4723909 5.370926  -7.9081032
 [3,]  0.420611907       0.320880918  0.099730989 -0.7472272 -0.7472272 6.390849  -4.7754160
 [4,]  0.298132442       0.135723778  0.162408664 -0.1049393 -0.1049393 7.294208  -0.7654494
```